code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_abilities.dm

```diff
@@ -0,0 +1,35 @@
+/datum/action/cooldown/mob_cooldown/ice_breathe
+	name = "Ice breathe"
+	desc = "Fire a cold line of fire towards the enemy!"
+	button_icon = 'icons/effects/magic.dmi'
+	button_icon_state = "fireball"
+	cooldown_time = 2 SECONDS
+	melee_cooldown_time = 0 SECONDS
+	click_to_activate = TRUE
+	///the range of fire
+	var/fire_range = 4
+
+/datum/action/cooldown/mob_cooldown/ice_breathe/Activate(atom/target_atom)
+	var/turf/target_fire_turf = get_ranged_target_turf_direct(owner, target_atom, fire_range)
+	var/list/burn_turfs = get_line(owner, target_fire_turf) - get_turf(owner)
+	dragon_fire_line(owner, burn_turfs, frozen = TRUE)
+	StartCooldown()
+	return TRUE
+
+/datum/action/cooldown/mob_cooldown/ice_breathe_all_directions
+	name = "Fire all directions"
+	desc = "Unleash lines of cold fire in all directions"
+	button_icon = 'icons/effects/fire.dmi'
+	button_icon_state = "1"
+	cooldown_time = 2 SECONDS
```

Comment:
If AI-controlled dragons can only use their omnidirectional breath every 3 normal breaths then this should have at least 3 times the cooldown in order to keep some kind of parity between AI and player control.
Bonus points if you just made "this unlocks after x attacks" be the way the ability works for everyone, but that's more complicated and I won't require it.